Objects/unicodeobject.c

```diff
@@ -6086,17 +6082,13 @@ _PyUnicode_DecodeUnicodeEscape(const char *s,
 
       error:
         endinpos = s-starts;
-        writer.min_length = end - s + writer.pos;
         if (unicode_decode_call_errorhandler_writer(
                 errors, &errorHandler,
                 "unicodeescape", message,
                 &starts, &end, &startinpos, &endinpos, &exc, &s,
                 &writer)) {
             goto onError;
         }
-        if (_PyUnicodeWriter_Prepare(&writer, writer.min_length, 127) < 0) {
-            goto onError;
```

Comment:
Of course. But isn't the widen done in `unicode_decode_call_errorhandler_writer`? When the error handler generates only one character, we don't need more space since we have already got enough space. But when the error handler generates more, `unicode_decode_call_errorhandler_writer` allocates the spaces for you. You did the change to `unicode_decode_call_errorhandler_writer` to avoid crash.